### PR TITLE
[core] Calculate GeoJSON tile geometries in a background thread

### PR DESCRIPTION
**Pull request by pozdnyakov**
Call `mapbox::geojsonvt::GeoJSONVT::getTile()` in a background thread, so that the rendering thread is not blocked.

**Before:**
before

**After:**

after

Fixes https://github.com/mapbox/mapbox-gl-native-team/issues/87

### CODE REVIEW

**Comment by pozdnyakov** [comment]
cc @LukasPaczos @Guardiola31337 @AnnaStaley

File: src/mbgl/style/sources/geojson_source_impl.cpp [hunk 1]
```diff
@@ -18,45 +19,39 @@ class GeoJSONVTData : public GeoJSONData {
...
-    }
+    Features getLeaves(const std::uint32_t, const std::uint32_t, const std::uint32_t) final { return {}; }
 
     std::uint8_t getClusterExpansionZoom(std::uint32_t) final {
         return 0;
     }
 
 private:
     mapbox::geojsonvt::GeoJSONVT impl;
+    SequencedScheduler scheduler;
```
Comment (alexshalamov) [inline src/mbgl/style/sources/geojson_source_impl.cpp]: why not to use dedicated thread pool? This will create new thread for each geojson source, right?

Comment (pozdnyakov) [reply src/mbgl/style/sources/geojson_source_impl.cpp]: we need a single thread that is keeping the state. Right, for each geojson source impl.

File: src/mbgl/tile/geometry_tile_worker.cpp [hunk 2]
```diff
@@ -170,6 +168,26 @@ void GeometryTileWorker::setLayers(std::vector<Immutable<LayerProperties>> layer
     }
 }
 
+void GeometryTileWorker::reset(uint64_t correlationID_) {
+    try {
```
Comment (tmpsantos) [inline src/mbgl/tile/geometry_tile_worker.cpp]: what exception this might throw?

Comment (alexshalamov) [reply src/mbgl/tile/geometry_tile_worker.cpp]: good point, we have quite a few of those in a tile worker.

Comment (pozdnyakov) [reply src/mbgl/tile/geometry_tile_worker.cpp]: I just took after other methods, no exceptions expected here :) I'll remove try.. catch

**Comment by pozdnyakov** [comment]
Invoking a thread per geojson source is not quite safe approach, as it can lead to OOM. Need to limit the possible amount of threads.

**Comment by pozdnyakov** [comment]
> Invoking a thread per geojson source is not quite safe approach, as it can lead to OOM. Need to limit the possible amount of threads.

Fixed with e6804088f0799f5405a0053ce34b97c183dce90a

**Comment by pozdnyakov** [comment]
**Note for change log:** Calculate GeoJSON tile geometries in a background thread, so that the rendering thread is not blocked.

**Pull request merged**